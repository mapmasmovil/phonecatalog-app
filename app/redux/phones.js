export const FETCH_PHONES = 'phones/FETCH_PHONES'
export const FETCH_PHONES_SUCCESS = 'phones/FETCH_PHONES_SUCCESS'
export const FETCH_PHONES_FAIL = 'phones/FETCH_PHONES_FAIL'


const initialState = {
  phones: [],
}

export default function reducer(state = initialState, action) {
  switch (action.type) {
    case FETCH_PHONES:
      return {
        ...state,
        loading: true,
      }

    case FETCH_PHONES_SUCCESS:
      return {
        ...state,
        loading: false,
        phones: action.payload.data.phones,
      }

    case FETCH_PHONES_FAIL:
      return {
        ...state,
        loading: false,
        error: 'Error while fetching phones'
      }

    default:
      return state
  }
}

export function fetchPhones() {
  return {
    type: FETCH_PHONES,
    payload: {
      request: {
        url: `phones/`
      }
    }
  }
}
