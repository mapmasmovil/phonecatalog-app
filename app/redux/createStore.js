import { createStore, applyMiddleware } from 'redux'
import { connect } from 'react-redux'
import axios from 'axios'
import axiosMiddleware from 'redux-axios-middleware'

import phones from 'masmovil-app/app/redux/phones'

export const API_URL = 'http://192.168.56.1:8000/'
const client = axios.create({
  baseURL: API_URL,
  responseType: 'json'
})

const store = createStore(
  phones,
  applyMiddleware(
    axiosMiddleware(client),
  )
)
export default store
