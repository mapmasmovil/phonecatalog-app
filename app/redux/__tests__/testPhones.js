import nock from 'nock'
import _ from 'lodash'

import {
  customCreateRealStore,
  timeout,
} from 'masmovil-app/app/jest/utils'
import * as phonesActions from 'masmovil-app/app/redux/phones'


const generateInitialState = () => ({
  phones: {
    entities: [],
  }
})

describe('Phones redux', () => {
  it('fetchPhones', async () => {
    const phonesMock = nock('http://fakeurl.fake')
      .get('/phones/')
      .query(true)
      .reply(200, {
        status: 200,
        phones: [
          {
            'id': 0,
            'title': 'iphone 5s',
          },
        ],
      })

    const store = customCreateRealStore(generateInitialState())
    await store.dispatch(phonesActions.fetchPhones())

    phonesMock.done()
    await timeout(20)

    expect(_.map(store.accumulator, a => a.type)).toEqual([
      "phones/FETCH_PHONES",
      "phones/FETCH_PHONES_SUCCESS",
    ])
    const state = store.getState()
    expect(state.phones.length).toEqual(1)
    expect(state).toMatchSnapshot()
  })
})