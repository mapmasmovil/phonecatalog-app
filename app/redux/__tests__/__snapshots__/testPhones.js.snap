// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Phones redux fetchPhones 1`] = `
Object {
  "loading": false,
  "phones": Array [
    Object {
      "id": 0,
      "title": "iphone 5s",
    },
  ],
}
`;
